crates/network-scanner-net/examples/tcp.rs

```diff
@@ -0,0 +1,63 @@
+use std::mem::MaybeUninit;
+use std::net::SocketAddr;
+
+use tokio::io::AsyncReadExt;
+use tokio::io::AsyncWriteExt;
+use trust_dns_resolver::config::*;
+use trust_dns_resolver::TokioAsyncResolver;
+
+#[tokio::main]
+pub async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::SubscriberBuilder::default()
+        .with_max_level(tracing::Level::TRACE)
+        .with_thread_names(true)
+        .init();
+
+    let async_runtime = network_scanner_net::runtime::Socket2Runtime::new()?;
+    let mut socket = async_runtime.new_socket(socket2::Domain::IPV4, socket2::Type::STREAM, None)?;
+    //create a tcp socket
+    let ip = {
+        let resolver = TokioAsyncResolver::tokio(ResolverConfig::default(), ResolverOpts::default());
+        let response = resolver.lookup_ip("info.cern.ch").await?;
+        let ip = response.iter().next().unwrap();
+        ip
+    };
+    tracing::info!("ip: {}", ip);
+    let socket_addr = SocketAddr::from((ip, 80));
+    let clone = socket_addr;
+    let handle = tokio::task::spawn(async move {
+        tracing::info!("connecting to cern using tokio");
+        let mut stream = tokio::net::TcpStream::connect(clone).await?; // making sure tokio io runtime does not collide with socket2 runtime
+        tracing::info!("conncted to cern using tokio");
+        let _ = stream.write(b"GET / HTTP/1.0\r\n\r\n").await?;
+        let buf = &mut [0u8; 1024];
+        let size = stream.read(buf).await?;
+        Ok::<Vec<_>, anyhow::Error>(buf[..size].to_vec())
+    }); // making sure tokio io event loop does not collide with socket2 event loop
```

Comment:
question: I don’t quite understand what is making sure there is no collision? Also, what is a collision in this case? The socket2 event loop is started in a different thread, so my understanding is that it is running in parallel without requiring any special handling. I understand what you are trying to achieve is called "concurrency" using async tasks. Is that correct?

Reply:
When I mentioned the possibility of a collision, I was referring to ensuring that there is no conflict in accessing the underlying IOCP/epoll or kqueue systems. Since both Mio and Polling essentially use the same system calls, I just want to make sure that two separate IO event loops can work in conjunction with two separate instances of the OS's asynchronous IO primitives without any issues.

Reply:
I see! Maybe you can you write this down in the comment, I think it would be useful to have the context

Reply:
updated comments